src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -942,37 +942,42 @@ window.qBittorrent.DynamicTable = (function() {
                     case "metaDL":
                     case "forcedMetaDL":
                         state = "downloading";
+                        img_path = "icons/downloading.svg";
                         break;
                     case "forcedUP":
                         state = "uploading";
+                        img_path = "icons/upload.svg";
                         break;
                     case "pausedDL":
-                        state = "media-playback-pause";
+                        state = "torrent-stop";
+                        img_path = "icons/torrent-stop.svg";
                         break;
                     case "pausedUP":
                         state = "checked-completed";
+                        img_path = "icons/checked-completed.svg";
                         break;
                     case "queuedDL":
                     case "queuedUP":
                         state = "queued";
+                        img_path = "icons/queued.svg";
                         break;
                     case "checkingDL":
                     case "checkingUP":
                     case "queuedForChecking":
                     case "checkingResumeData":
                     case "moving":
                         state = "force-recheck";
+                        img_path = "icons/force-rechek.svg";
                         break;
                     case "unknown":
                     case "missingFiles":
                         state = "error";
+                        img_path = "icons/error.svg";
                         break;
                     default:
                         break; // do nothing
                 }
 
-                const img_path = 'icons/' + state + '.svg';
```

Comment:
Now that each state is assigned an icon explicitly, it seems that all state  cases now have to be added to the switch above from the list in the link below . I made and tested the following diff, i hope i didn't miss any state:

https://github.com/qbittorrent/qBittorrent/blob/2071ec0c968031d93c5ae77beba19ad992e98ba8/src/webui/api/serialize/serialize_torrent.cpp#L48



```
--- a/src/webui/www/private/scripts/dynamicTable.js
+++ b/src/webui/www/private/scripts/dynamicTable.js
@@ -936,18 +936,29 @@ window.qBittorrent.DynamicTable = (function() {
             // state_icon
             this.columns['state_icon'].updateTd = function(td, row) {
                 let state = this.getRowValue(row);
+                let img_path;
                 // normalize states
                 switch (state) {
                     case "forcedDL":
                     case "metaDL":
                     case "forcedMetaDL":
+                    case "downloading":
                         state = "downloading";
                         img_path = "icons/downloading.svg";
                         break;
                     case "forcedUP":
+                    case "uploading":
                         state = "uploading";
                         img_path = "icons/upload.svg";
                         break;
+                    case "stalledUP":
+                        state = "stalledUP";
+                        img_path = "icons/stalledUP.svg";
+                        break;
+                    case "stalledDL":
+                        state = "stalledDL";
+                        img_path = "icons/stalledDL.svg";
+                        break;
                     case "pausedDL":
                         state = "torrent-stop";
                         img_path = "icons/torrent-stop.svg";
@@ -967,7 +978,7 @@ window.qBittorrent.DynamicTable = (function() {
                     case "checkingResumeData":
                     case "moving":
                         state = "force-recheck";
-                        img_path = "icons/force-rechek.svg";
+                        img_path = "icons/force-recheck.svg";
                         break;
                     case "unknown":
                     case "missingFiles":

```

p.s. the re-assignment of "state" variable in the cases seem to only be used for icon tooltip in case you want to change any of them.

Reply:
Do I need to change anything at, `src/webui/api/serialize/serialize_torrent.cpp#L48`?

Reply:
>Do I need to change anything at, src/webui/api/serialize/serialize_torrent.cpp#L48?

No.

Reply:
There's a small problem with some of these lines indentation see https://github.com/qbittorrent/qBittorrent/runs/7573296595?check_suite_focus=true and after that, PR LGTM.

Reply:
Done.